Single Random Data Generator

Giolno opened the pull request:
Still working on new features.

Signed-off-by: Zongyu zy728@york.ac.uk

Nniy committed:
Add Single Random Data Generator

Still working on new features.

Signed-off-by: Zongyu <zy728@york.ac.uk>

review-ninja commented:
<a href="https://app.review.ninja/Llama-Group/Project-Llama/pull/108" target="_blank"><img src="https://app.review.ninja/63898156/pull/108/badge" alt="ReviewNinja"/>

ghost commented:
@al1enSuu @BlueCocoa 
Plz review this

BlueCocoa commented on llama_cpp/src/Utility/DataGenerator.h:
```diff
@@ -25,20 +25,16 @@
- public:
-    ContinuousData() {}
-
+    void generateRandomData(std::vector<T> *targetVector, int count = 10);
```
Maybe we should make the default value more configurable by using macro.

BlueCocoa commented on llama_cpp/src/Utility/DataGenerator.h:
```diff
@@ -25,20 +25,16 @@
-
+    void generateRandomData(std::vector<T> *targetVector, int count = 10);
+    template<typename T>
+    void generateContinuousData(std::vector<T> *targetVector, int count = 10, bool reverse = false);
```
IMO, since we don't hold any data in this class, perhaps it would be nicer if all the methods are class method.

BlueCocoa commented on llama_cpp/src/Utility/DataGenerator.cpp:
```diff
@@ -103,3 +103,44 @@ void llama::ContinuousData::generateContinuousData<std::string>
+void llama::DataGenerator::generateSingleData<std::string>
+(std::string *targetData, std::string *givenData, int caseSwitch) {
+    static const char alpha[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    "§1234567890-=[]\';\\,./`±!@£$^&*()_+{}:\"|<>?~";
```
It's unnecessary to add a `\` before the `'`

BlueCocoa commented on llama_cpp/src/Utility/DataGenerator.cpp:
```diff
@@ -103,3 +103,44 @@ void llama::ContinuousData::generateContinuousData<std::string>
 }
+
+template<>
+void llama::DataGenerator::generateSingleData<int>(int *targetData, int *givenData, int caseSwitch) {
```
givenData is unused?

Nniy committed:
Added Data Comparing Feature

Added data comparing feature, but still need to be optimized.

Signed-off-by: Zongyu <zy728@york.ac.uk>

0xsuu committed:
Merge branch 'master' into feature/SingleData

And fix conflict.

Signed-off-by: al1ensuu <462993145@qq.com>

Nniy committed:
Merge From master

Merge from master

Signed-off-by: Zongyu <zy728@york.ac.uk>

Nniy committed:
Fix Conflicts

Merge remote-tracking branch 'origin/feature/SingleData' into feature/SingleData

Signed-off-by: Zongyu <zy728@york.ac.uk>

0xsuu committed:
Merge remote-tracking branch 'origin/master' into feature/SingleData

0xsuu committed:
Update comparison with single datum generator

Signed-off-by: al1ensuu <462993145@qq.com>

BlueCocoa committed:
A few improvements

- Remove redundant include in DataGenerator.cpp since we include
headers in DataGenerator.h
- Using generateRandomDataFromSet as a wrapper for
generateRandomData<std::string>
- Using macro DATA_GENERATOR_DEFAULT_COUNT
- Using macro DATA_GENERATOR_DEFAULT_STRING_MAX_LENGTH
- Make all methods class method (corresponding changes already applied
to example file)

0xsuu committed:
Merge remote-tracking branch 'origin/feature/SingleData' into feature/SingleData

Signed-off-by: al1ensuu <462993145@qq.com>

0xsuu committed:
Merge branch 'feature/SingleData' of https://github.com/Llama-Group/Project-Llama into feature/SingleData

BlueCocoa committed:
Align code

0xsuu commented:
Please rebuild.

Nniy committed:
Debug SingleData

Debug singleData.

Signed-off-by: Zongyu <zy728@york.ac.uk>

Nniy committed:
Add Exceptions

Added exceptions to warning if the the given data is already maximum or minimum.

Signed-off-by: Zongyu <zy728@york.ac.uk>

Nniy committed:
Fix Bugs

Fixed bugs.

Signed-off-by: Zongyu <zy728@york.ac.uk>

0xsuu committed:
Fix Alex's super stupid bug

Don't assign int with double, hahahhahahahahahahah

Signed-off-by: al1ensuu <462993145@qq.com>

0xsuu committed:
Fix some nits

Causing lint error and travis fail to build.

Signed-off-by: al1ensuu <462993145@qq.com>

0xsuu committed:
Add NE for Sort

Signed-off-by: al1ensuu <462993145@qq.com>

0xsuu committed:
Merge remote-tracking branch 'origin/master' into feature/SingleData

0xsuu committed:
Refactoring utilut add DG UT

We are doing too much fucking things on this branch! Should ignore coverall coverage!

Signed-off-by: al1ensuu <462993145@qq.com>

0xsuu committed:
Add NE test and use static methods of DG

Signed-off-by: al1ensuu <462993145@qq.com>

0xsuu committed:
Add exception test

Whooray! go to sleep now! leave shitty thing to Alex!

Signed-off-by: al1ensuu <462993145@qq.com>

0xsuu committed:
Add minor massive to speed up Sort test

Cuz we have so many dumbass sorts!

Signed-off-by: al1ensuu <462993145@qq.com>

0xsuu committed:
Style fixes

Signed-off-by: al1ensuu <462993145@qq.com>

0xsuu committed:
Remove coverage for UT

Signed-off-by: al1ensuu <462993145@qq.com>

0xsuu commented:
LGTM
